go/def.bzl

```diff
@@ -581,19 +581,36 @@ def go_test_impl(ctx):
   lib_result = go_library_impl(ctx)
   main_go = ctx.outputs.main_go
   prefix = _go_prefix(ctx)
-
   go_import = _go_importpath(ctx)
 
-  args = (["--package", go_import, "--output", ctx.outputs.main_go.path] +
-          [i.path for i in lib_result.go_sources])
-
-  inputs = list(lib_result.go_sources) + list(ctx.files.toolchain)
+  cmds = [
```

Comment:
I agree that would be mostly simpler, though it might be more opaque for quoted lists of items like `UNFILTERED_TEST_FILES`.

In the not-too-distant future, I'd like to rewrite all of this in Go, since the Bash code is difficult to work with (hard to read, hard to test, error-prone). For now, I'll leave it as is for consistency with other blocks of commands formatted the same way.